pkg/stacks/unpack.go

```diff
@@ -156,52 +170,50 @@ type StackPackage struct {
 	// appSet indicates if a App has been assigned through SetApp (for use by GotApp)
 	appSet bool
 
+	// behaviorSet indicates if a Behavior has been assigned through SetBehavior (for use by GotBehavior)
+	behaviorSet bool
+
 	// baseDir is the directory that serves as the base of the stack package (it should be absolute)
 	baseDir string
+
+	// tmplCtrlImage is the Template Controller image to handle template stacks
+	tmplCtrlImage string
 }
 
 // Yaml returns a multiple document YAML representation of the Stack Package
 // This YAML includes the Stack itself and and all CRDs managed by that Stack.
 func (sp *StackPackage) Yaml() (string, error) {
-	var builder strings.Builder
-
+	builder := &strings.Builder{}
 	builder.WriteString(yamlSeparator)
 
 	// For testing, we want a predictable output order for CRDs
 	orderedKeys := orderStackCRDKeys(sp.CRDs)
 
 	for _, k := range orderedKeys {
 		crd := sp.CRDs[k]
-		b, err := yaml.Marshal(crd)
-		if err != nil {
-			return "", errors.Wrap(err, fmt.Sprintf("could not marshal CRD (%s)", k))
+		if err := writeYaml(builder, crd, "CRD"); err != nil {
+			return "", err
 		}
-		builder.Write(b)
-		builder.WriteString(yamlSeparator)
-	}
-
-	b, err := yaml.Marshal(sp.Stack)
-	if err != nil {
-		return "", errors.Wrap(err, "could not marshal Stack")
 	}
 
-	if _, err := builder.Write(b); err != nil {
-		return "", errors.Wrap(err, "could not write YAML output to buffer")
-	}
+	if sp.GotBehavior() {
+		sp.Stack.DeepCopyIntoStackDefinition(&sp.StackDefinition)
 
-	// Behaviors are optional, so we skip it if it doesn't exist
-	if len(sp.Behavior.Object) > 0 {
-		bb, err := yaml.Marshal(sp.Behavior.Object)
-		if err != nil {
-			return "", errors.Wrap(err, "could not marshal behavior")
+		// New Format, using 'behavior.yaml'
```

Comment:
wondering if we should be versioning our stack package format now, so it's clear to a parser which format they should expect to be parsing, instead of figuring it out more heuristically/implicitly like we are doing here.